### ISSUE TITLE
Should `BoutDataset.save()` record the version of `xBOUT` used to produce the files?

### ISSUE THREAD

**Issue by johnomotani** [boutproject/xBOUT#244]
Might be useful for provenance tracking and reproducibility to know what version of `xBOUT` was used to produce some file. Similar to https://github.com/boutproject/boutdata/pull/79.

**Comment by ZedThree** [comment]
Oh, I perhaps should've thought about this on that other PR, it would be good to have this set of metadata as attributes:

- title
- software_name
- software_version
- netcdf_version
- date_created
- id

I've cobbled this set together from various conventions, including [netCDF][netcdf] and the [Climate Forcasting][cf] conventions, and I've been trying to get it into various output files.

I think we might already have some of these?

  !> [netcdf]: https://docs.unidata.ucar.edu/netcdf-c/current/attribute_conventions.html
  !> [cf]: https://cfconventions.org

**Comment by bendudson** [comment]
Quick bikeshedding question: Should `software_version` be a semantic or at least sequential version number, or a git checksum, or a combination?

**Comment by ZedThree** [comment]
Yes :)

Personally, I'm a big fan of [setuptools_scm](https://github.com/pypa/setuptools_scm#default-versioning-scheme)'s scheme, which is SemVer compatible. It's basically:

- no distance and clean:
    `{tag}`
- distance and clean:
    `{next_version}`.dev`{distance}`+`{scm letter}{revision hash}`

So a release would be `1.2.3` say, and the first commit after that would get version `1.2.3.dev1+gdeadbeef`

**Comment by johnomotani** [comment]
:+1: I was gonna say what @ZedThree said.

**Comment by dschwoerer** [comment]
Offtopic: Isn't the next one `1.2.4.dev1+gdeadbeef`?

**Comment by ZedThree** [comment]
Good catch, yes. I think there's some setting to make it bump the minor version instead too.
